components/lib/crash/src/main/java/mozilla/components/lib/crash/BreadcrumbArrayList.kt

```diff
@@ -0,0 +1,20 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package mozilla.components.lib.crash
+
+import mozilla.components.support.base.crash.Breadcrumb
+
+internal class BreadcrumbArrayList(
+    private val maxSize: Int
+) : ArrayList<Breadcrumb>() {
+    @Synchronized
+    override fun add(element: Breadcrumb): Boolean {
+        if (this.size >= maxSize) {
+            this.removeAt(0)
+        }
+
+        return super.add(element)
```

Comment:
Extending `ArrayList` may be overkill here, if we can have this add/remove call inside `CrashReporter`? :)

Technically calling things like `addAll()` on this separate class will not honor the limit.

Reply:
Agreed.  This is probably a overkill.  Sure, I'll move this into the `crashreporter`